app/src/main/java/org/groundplatform/android/MainActivity.kt

```diff
@@ -238,13 +246,10 @@ class MainActivity : AbstractActivity() {
     }
   }
 
-  private fun handleDeepLinkIntent(intent: Intent) {
-    val viewModel = viewModelFactory[this, SurveySelectorViewModel::class.java]
-    intent.data?.let { uri ->
+  private fun NavController.handleDeepLinkIfNeeded(uri: Uri) {
+    if (uri.host == "groundplatform.org" && uri.pathSegments.firstOrNull() == "survey") {
```

Comment:
We can't hard-code the host here since it will be different for different deployments. Is checking the host necessary at all? Iiuc the app will only receive deep links for the host defined in `AndroidManifest.xml`, no?

Ok to check the path segment, in case we support other paths in the future, but please move to a string const `SURVEY_PATH_SEGMENT`.

Reply:
Fixed